### DISCUSSION TITLE
Exclude njk file dynamically from build

### DISCUSSION THREAD

**Discussion by jens-struct** [11ty/eleventy#2285] [Q&A] [answered]
Is it possible to exclude njk files dynamically from build?

I know you can set permalink to false, so these njk files aren't written to output, but the files are nevertheless executed/parsed.
I want to disable the latter dynamically, so i can reduce the build time under certain conditions (environments).

**Comment by pdehaan** [comment, marked as answer]
@jens-hmmh I think I'm a bit confused by the workflow. If you don't want to output the Nunjucks templates (`permalink: false`) does it work better if you just move the templates into the _includes/ directory?

I think you're correct that 11ty will still parse files regardless of whether they set `permalink: false` or `eleventyExcludeFromCollections: true` (which can be a useful way of preventing files from being written, but still adding them to collections or building 1 page sites from multiple templates).

Or if you want to prevent 11ty from globally parsing ANY .njk files, I wonder if you could just do something with https://www.11ty.dev/docs/config/#template-formats. You could even do a check for if `process.env.NODE_ENV === "production"` and optionally ignore certain templates to speed up dev builds.

**Reply by cfjedimaster** [reply]
Oh oh - what about the new ability to dynamically set ignore settings? If so, they could use that to ignore 1 or more that way.

**Reply by jens-struct** [reply]
Thanks for the answer!

I just want to exclude some njk files (or url's), not all (under certain conditions). And save compile & write time (and also to prevent compile errors if the data which is used in these excluded njk files doesn't exist). For now i included safeguards in these njk files, which check if the data exists. Which works good enough. So it is not that big of a problem for me, that i can't disable parsing.

I also looked into the nunjucks environment, if there is an option to exclude files, but i found no option that could do that.

Will have a closer look into eleventy serverless, seems that could be an option.

I'm working on some kind of a preview build. So a short build time is important in this case. (1500+ written files, if all templates are used/compiled, 30s build time)

**Reply by jens-struct** [reply]
> Oh oh - what about the new ability to dynamically set ignore settings? If so, they could use that to ignore 1 or more that way.

I tried that, but njk files can't be ignored. The ignore docs mention this. Or i did/understood that wrong.

**Reply by cfjedimaster** [reply]
Where do you see that?

**Reply by jens-struct** [reply]
I understood that "Layouts, include files, extends, partials, macros, and other lower level template features aren’t relevant to this feature." sentence wrong. It works! I had the globbing wrong, first time i tried.

Ignoring the njk files omits parsing them.
